### PR TITLE
[ADD] mail: blacklist mechanism

### PR DESCRIPTION
**Pull request by mart-e**
WIP

Task: https://www.odoo.com/web#id=23876&view_type=form&model=project.task&action=333&active_id=248&menu_id=4720

### CODE REVIEW

File: addons/mail/models/mail_thread.py [hunk 1]
```diff
@@ -1726,6 +1726,12 @@ def message_post(self, body='', subject=None, message_type='notification',
         if author_id is None:  # keep False values
             author_id = self.env['mail.message']._get_default_author().id
 
+        author = self.env['res.partner'].browse(author_id)
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_thread.py]: You should also check email_from if author_id is not set.

File: addons/mail/models/mail_channel.py [hunk 2]
```diff
@@ -575,3 +575,11 @@ def channel_fetch_preview(self):
             del(channel['message_id'])
             channel['last_message'] = message
         return channels_preview.values()
+
+
+class MailBlacklist(models.Model):
```
Comment (tde-banana-odoo) [inline addons/mail/models/mail_channel.py]: Should'nt we put it into its own file ?

File: addons/mail/tests/test_mail_gateway.py [hunk 3]
```diff
@@ -717,3 +717,11 @@ def test_private_discussion(self):
 
             self.assertEqual(msg_fw.model, 'mail.channel')
             self.assertFalse(msg_fw.parent_id)
+
+    @mute_logger('openerp.addons.mail.models.mail_thread', 'openerp.models', 'openerp.addons.mail.models.mail_mail')
+    def test_blacklist(self):
+        """ Testing blacklisted user. """
+        msg1_pids = [self.env.user.partner_id.id, self.partner_1.id]
+        # Raoul has been naughty and is blacklisted
```
Comment (tde-banana-odoo) [inline addons/mail/tests/test_mail_gateway.py]: naughty Raoul !

**Comment by treviser** [comment]
@mart-e what a fascinating repository name :-)
SCNR

**Comment by mart-e** [comment]
not a wished feature

**Pull request closed without merging**